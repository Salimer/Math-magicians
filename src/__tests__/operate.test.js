import operate from '../logic/operate';

describe('operate function', () => {
  test('performs addition correctly', () => {
    const result = operate('2', '3', '+');
    expect(result).toMatchSnapshot();
  });

  test('performs subtraction correctly', () => {
    const result = operate('5', '2', '-');
    expect(result).toMatchSnapshot();
  });

  test('performs multiplication correctly', () => {
    const result = operate('4', '6', 'x');
    expect(result).toMatchSnapshot();
  });

  test('performs division correctly', () => {
    const result = operate('10', '2', 'รท');
    expect(result).toMatchSnapshot();
  });
});
