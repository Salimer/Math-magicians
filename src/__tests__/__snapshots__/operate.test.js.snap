// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`operate function performs addition correctly 1`] = `"5"`;

exports[`operate function performs division correctly 1`] = `"5"`;

exports[`operate function performs multiplication correctly 1`] = `"24"`;

exports[`operate function performs subtraction correctly 1`] = `"3"`;
